src/unitxt/inference.py

```diff
@@ -809,16 +855,18 @@ class WMLInferenceEngine(
     parameters: Optional[WMLInferenceEngineParams] = None
     concurrency_limit: int = 10
     _client: Any = InternalField(default=None, name="WML client")
+    _model: Any = InternalField(default=None, name="WML model")
 
     def verify(self):
         super().verify()
 
-        if self.credentials is not None:
-            for key in self.credentials:
-                if key not in ["url", "apikey", "project_id", "space_id"]:
-                    raise ValueError(
-                        f'Illegal credential key: {key}, use only ["url", "apikey", "project_id", "space_id"]'
-                    )
+        assert (
```

Comment:
Are there no special keys in wml with more than 10?

Reply:
Unfortunately 10 is the limit imposed by `generate` method of `ModelInference`. I put this check in unitxt as well so that users are aware of it